addons/web/static/src/js/views/search_panel.js

```diff
@@ -1,788 +1,221 @@
-     * @param {Object} params.sections
-     * @param {Object} [params.state] state exported by another searchpanel
-     *   instance
-     */
-    init(parent, params) {
-        this._super(...arguments);
-
-        this.categories = {};
-        this.filters = {};
-        for (const section of Object.values(params.sections)) {
-            const key = section.type === 'category' ? 'categories' : 'filters';
-            this[key][section.id] = section;
-        }
-
-        this.initialState = params.state;
-        this.scrollTop = this.initialState && this.initialState.scrollTop || null;
-        this.defaultValues = params.defaultValues || {};
-        if (params.defaultNoFilter) {
-            Object.values(this.categories).forEach(({ fieldName }) => {
-                this.defaultValues[fieldName] = this.defaultValues[fieldName] || false;
+     * Search panel
+     *
+     * Represent an extension of the search interface located on the left side of
+     * the view. It is divided in sections defined in a "<searchpanel>" node located
+     * inside of a "<search>" arch. Each section is represented by a list of different
+     * values (categories or ungrouped filters) or groups of values (grouped filters).
+     * Its state is directly affected by its model (@see SearchPanelModel).
+     * @extends Component
+     */
+    class SearchPanel extends Component {
+        constructor() {
+            super(...arguments);
+
+            useSubEnv({
+                searchModel: this.props.searchModel,
             });
-        }
-        this.fields = params.fields;
-        this.model = params.model;
-        this.className = params.classes.concat(['o_search_panel']).join(' ');
-        this.reloadOptions = null;
-        this.searchDomain = params.searchDomain;
-        this.viewDomain = params.viewDomain;
-    },
-    /**
-     * @override
-     */
-    async willStart() {
-        const _super = this._super;
-        if (this.initialState) {
-            this.filters = this.initialState.filters;
-            this.categories = this.initialState.categories;
-        } else {
-            await this._fetchCategories(true);
-            await this._fetchFilters();
-            await this._applyDefaultFilterValues();
-        }
-        return _super.call(this, ...arguments);
-    },
-    /**
-     * @override
-     */
-    start() {
-        this._render();
-        return this._super(...arguments);
-    },
-    /**
-     * Called each time the searchPanel is attached into the DOM.
-     */
-    on_attach_callback() {
-        if (this.scrollTop !== null) {
-            this.el.scrollTop = this.scrollTop;
-        }
-    },
 
-    //-------------------------------------------------------------------------
-    // Public
-    //-------------------------------------------------------------------------
+            this.state = useState({
+                expanded: {},
```

Comment:
expanded should be exported too maybe

Reply:
--> keep only temp --> move expanded and scrollTop in cpme we want to export/import them

Reply:
nope